dev/io.openliberty.reporting/src/io/openliberty/reporting/CVEServiceClient.java

```diff
@@ -0,0 +1,173 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package io.openliberty.reporting;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.ConnectException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import com.ibm.json.java.JSONObject;
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+/**
+ * <p>
+ * The CVE Service Client makes the connection to the cloud service and then
+ * gets back the response containing a JSON Array which contains the list of
+ * CVE's that could potentially have impact.
+ * </p>
+ */
+public class CVEServiceClient {
+
+	private static final TraceComponent tc = Tr.register(CVEServiceClient.class);
+
+	/**
+	 * <p>
+	 * Retrieves the CVE Data from the cloud service.
+	 * </p>
+	 * 
+	 * @param data    Map<String, String>
+	 * @param urlLink URL link which is set in the Server.xml
+	 * @return json JSONObject
+	 * @throws IOException
+	 */
+	public JSONObject retrieveCVEData(Map<String, String> data, String urlLink) throws IOException {
+
+		String jsonData = buildJsonString(data);
+
+		if (!urlLink.startsWith("https")) {
+			throw new MalformedURLException("Invalid Protocol, expected \"https\': " + urlLink);
+		}
+
+		URL url = new URL(urlLink);
+
+		HttpsURLConnection connection = getConnection(url);
+		if (connection != null) {
+			sendData(connection, jsonData);
+		}
+
+		if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+			Tr.debug(this, tc, jsonData);
+		}
+
+		JSONObject json = JSONObject.parse(getResponse(connection));
```

Comment:
```suggestion
		try (Reader reader = new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8)) {
			JSONObject json = JSONObject.parse(reader);
		}
```

Can get rid of the getResponse method.